src/docfx/build/redirection/RedirectionProvider.cs

```diff
@@ -178,8 +190,26 @@ private static RedirectionItem[] LoadRedirectionModel(ErrorBuilder errors, strin
                     // Rebase source_path based on redirection definition file path
                     var basedir = Path.GetDirectoryName(fullPath) ?? "";
 
-                    var results = new List<RedirectionItem>();
-                    foreach (var item in redirections.Concat(renames))
+                    var allRedirections = redirections.Concat(renames);
+
+                    if (isSubRedirectionFile && repository != null)
+                    {
+                        var prefix = Path.GetDirectoryName(Path.GetRelativePath(repository.Path, fullPath));
+
+                        if (!string.IsNullOrEmpty(prefix))
+                        {
+                            allRedirections = allRedirections.Select(x =>
+                            {
+                                if (!x.SourcePath.IsDefault)
+                                {
+                                    x.SourcePath = new PathString(Path.Combine(prefix, x.SourcePath));
+                                }
+                                return x;
+                            }).ToImmutableArray();
```

Comment:
what is `ToImmutableArray` used for? It seems like `allRedirections` is just used for looping?

Reply:
Removed.